core/src/main/java/hudson/model/labels/LabelExpression.java

```diff
@@ -210,4 +225,156 @@ public LabelOperatorPrecedence precedence() {
             return LabelOperatorPrecedence.IMPLIES;
         }
     }
+
+    //region Auto-Completion and Validation
+
+    /**
+     * Utility class for taking the current input value and computing a list of potential terms to match against the
+     * list of defined labels.
+     */
+    static class AutoCompleteSeeder {
```

Comment:
```suggestion
    static class LabelAutoCompleteSeeder {
```

Reply:
Yes, if moving it out of `LabelExpression`, adding the `Label` prefix makes total sense.